README.md
# Megalabs Tokens

> Design variables published as Swift and Kotlin.

| Platform | Languages | Package |
| --- | --- | --- |
| iOS | Swift | [![CocoaPods](https://img.shields.io/cocoapods/v/MegalabsTokens)](https://cocoapods.org/pods/MegalabsTokens) |
| Android | Kotlin | [![JitPack](https://img.shields.io/jitpack/v/github/megalabs/megalabs-tokens)](https://jitpack.io/#thumbtack/megalabs-tokens) |

## Releasing

1. Run `git checkout master && git pull` to get the latest changes.
2. Add the new version number to the `CHANGELOG.md`.
3. Commit the changes on `master` with `git add CHANGELOG.md && git commit -m "Preparing for release."`
4. Run `yarn release`.

CHANGELOG.md
# Changelog

## 0.1.0 - 2018-11-27

### Added

-   Added initial color tokens

tokens.js
module.exports = [
    {
        name: 'Color',
        description: 'Megafon Bank Colors',
        tokens: [
            {
                id: 'green',
                platforms: {
                    ios: {
                        name: 'green',
                        description: 'Green #00B956',
                        value:
                            'UIColor = UIColor(red: 0, green: 0.725, blue: 0.337, alpha: 1)',
                    },
                    android: {
                        name: 'mfb_green',
                        value: '#00B956',
                    },
                },
                format: 'color',
                type: 'string',
            },
            {
                id: 'green-glow',
                platforms: {
                    ios: {
                        name: 'greenglow',
                        description: 'Green Glow #00D463',
                        value:
                            'UIColor(red: 0, green: 0.831, blue: 0.388, alpha: 1)',
                    },
                    android: {
                        name: 'mfb_greenglow',
                        value: '#00D463',
                    },
                },
                format: 'color',
                type: 'string',
            },
            {
                id: 'green-dark',
                platforms: {
                    ios: {
                        name: 'greendark',
                        description: 'Green Dark #00A14B',
                        value:
                            'UIColor(red: 0, green: 0.631, blue: 0.294, alpha: 1)',
                    },
                    android: {
                        name: 'mfb_greendark',
                        value: '#00A14B',
                    },
                },
                format: 'color',
                type: 'string',
            },
        ],
    },
];

src/MegafonTokens.js
module.exports = [
    {
        name: 'Color',
        description: 'Megafon Bank Colors',
        tokens: [
            {
                id: 'brand-green',
                platforms: {
                    ios: {
                        name: 'brandGreen',
                        description: 'Brand Green #00B956',
                        value:
                            'UIColor = UIColor(red: 0, green: 0.725, blue: 0.337, alpha: 1.0)',
                    },
                    android: {
                        name: 'brand_green',
                        value: '#00B956',
                    },
                },
                format: 'color',
                type: 'string',
            },
            {
                id: 'brand-green-glow',
                platforms: {
                    ios: {
                        name: 'brandGreenGlow',
                        description: 'Brand Green Glow #00D463',
                        value:
                            'UIColor = UIColor(red: 0, green: 0.831, blue: 0.388, alpha: 1.0)',
                    },
                    android: {
                        name: 'brand_green_glow',
                        value: '#00D463',
                    },
                },
                format: 'color',
                type: 'string',
            },
            {
                id: 'brand-green-dark',
                platforms: {
                    ios: {
                        name: 'brandGreenDark',
                        description: 'Brand Green Dark #00A14B',
                        value:
                            'UIColor = UIColor(red: 0, green: 0.631, blue: 0.294, alpha: 1)',
                    },
                    android: {
                        name: 'brand_green_dark',
                        value: '#00A14B',
                    },
                },
                format: 'color',
                type: 'string',
            },
            {
                id: 'brand-purple',
                platforms: {
                    ios: {
                        name: 'brandPurple',
                        description: 'Brand Purple #731982',
                        value:
                            'UIColor = UIColor(red: 0.451, green: 0.098, blue: 0.51, alpha: 1.0)',
                    },
                    android: {
                        name: 'brand_purple',
                        value: '#731982',
                    },
                },
                format: 'color',
                type: 'string',
            },
            {
                id: 'brand-purple-glow',
                platforms: {
                    ios: {
                        name: 'brandPurpleGlow',
                        description: 'Brand Purple Glow #891E9C',
                        value:
                            'UIColor = UIColor(red: 0.537, green: 0.118, blue: 0.612, alpha: 1.0)',
                    },
                    android: {
                        name: 'brand_purple_glow',
                        value: '#891E9C',
                    },
                },
                format: 'color',
                type: 'string',
            },
            {
                id: 'brand-purple-dark',
                platforms: {
                    ios: {
                        name: 'brandPurpleDark',
                        description: 'Brand Purple Dark #5C1469',
                        value:
                            'UIColor = UIColor(red: 0.361, green: 0.078, blue: 0.412, alpha: 1.0)',
                    },
                    android: {
                        name: 'brand_purple_dark',
                        value: '#5C1469',
                    },
                },
                format: 'color',
                type: 'string',
            },
            {
                id: 'brand-tangerine',
                platforms: {
                    ios: {
                        name: 'brandTangerine',
                        description: 'Brand Tangerine #FFA717',
                        value:
                            'UIColor = UIColor(red: 1.0, green: 0.655, blue: 0.09, alpha: 1.0)',
                    },
                    android: {
                        name: 'brand_tangerine',
                        value: '#FFA717',
                    },
                },
                format: 'color',
                type: 'string',
            },
            {
                id: 'brand-blue',
                platforms: {
                    ios: {
                        name: 'brandBlue',
                        description: 'Brand Blue #1690FC',
                        value:
                            'UIColor = UIColor(red: 0.086, green: 0.565, blue: 0.988, alpha: 1.0)',
                    },
                    android: {
                        name: 'brand_blue',
                        value: '#1690FC',
                    },
                },
                format: 'color',
                type: 'string',
            },
            {
                id: 'brand-blush',
                platforms: {
                    ios: {
                        name: 'brandBlush',
                        description: 'Brand Blush #E8516D',
                        value:
                            'UIColor = UIColor(red: 0.91, green: 0.318, blue: 0.427, alpha: 1.0)',
                    },
                    android: {
                        name: 'brand_blush',
                        value: '#E8516D',
                    },
                },
                format: 'color',
                type: 'string',
            },
            {
                id: 'graphite',
                platforms: {
                    ios: {
                        name: 'graphite',
                        description: 'Graphite #333333',
                        value:
                            'UIColor = UIColor(red: 0.2, green: 0.2, blue: 0.2, alpha: 1.0)',
                    },
                    android: {
                        name: 'graphite',
                        value: '#333333',
                    },
                },
                format: 'color',
                type: 'string',
            },
            {
                id: 'haze',
                platforms: {
                    ios: {
                        name: 'haze',
                        description: 'Haze #989898',
                        value:
                            'UIColor = UIColor(red: 0.596, green: 0.596, blue: 0.596, alpha: 1.0)',
                    },
                    android: {
                        name: 'haze',
                        value: '#989898',
                    },
                },
                format: 'color',
                type: 'string',
            },
            {
                id: 'smoke',
                platforms: {
                    ios: {
                        name: 'smoke',
                        description: 'Smoke #ADADAD',
                        value:
                            'UIColor = UIColor(red: 0.678, green: 0.678, blue: 0.678, alpha: 1.0)',
                    },
                    android: {
                        name: 'smoke',
                        value: '#ADADAD',
                    },
                },
                format: 'color',
                type: 'string',
            },
            {
                id: 'dust',
                platforms: {
                    ios: {
                        name: 'dust',
                        description: 'Dust #D8D8D8',
                        value:
                            'UIColor = UIColor(red: 0.847, green: 0.847, blue: 0.847, alpha: 1.0)',
                    },
                    android: {
                        name: 'dust',
                        value: '#D8D8D8',
                    },
                },
                format: 'color',
                type: 'string',
            },
            {
                id: 'ash',
                platforms: {
                    ios: {
                        name: 'ash',
                        description: 'Ash #E5E5E5',
                        value:
                            'UIColor = UIColor(red: 0.9, green: 0.9, blue: 0.9, alpha: 1.0)',
                    },
                    android: {
                        name: 'ash',
                        value: '#E5E5E5',
                    },
                },
                format: 'color',
                type: 'string',
            },
            {
                id: 'mist',
                platforms: {
                    ios: {
                        name: 'mist',
                        description: 'Mist #F0F0F0',
                        value:
                            'UIColor = UIColor(red: 0.942, green: 0.942, blue: 0.942, alpha: 1.0)',
                    },
                    android: {
                        name: 'mist',
                        value: '#F0F0F0',
                    },
                },
                format: 'color',
                type: 'string',
            },
            {
                id: 'super-glue',
                platforms: {
                    ios: {
                        name: 'superGlue',
                        description: 'Super glue #F4F4F4',
                        value:
                            'UIColor = UIColor(red: 0.956, green: 0.956, blue: 0.956, alpha: 1.0)',
                    },
                    android: {
                        name: 'super_glue',
                        value: '#F4F4F4',
                    },
                },
                format: 'color',
                type: 'string',
            },
            {
                id: 'glue',
                platforms: {
                    ios: {
                        name: 'glue',
                        description: 'Glue #F7F7F7',
                        value:
                            'UIColor = UIColor(red: 0.969, green: 0.969, blue: 0.969, alpha: 1.0)',
                    },
                    android: {
                        name: 'glue',
                        value: '#F7F7F7',
                    },
                },
                format: 'color',
                type: 'string',
            },
            {
                id: 'black',
                platforms: {
                    ios: {
                        name: 'black',
                        description: 'Black #000000',
                        value:
                            'UIColor = UIColor(red: 0.0, green: 0.0, blue: 0.0, alpha: 1.0)',
                    },
                    android: {
                        name: 'black',
                        value: '#000000',
                    },
                },
                format: 'color',
                type: 'string',
            },
            {
                id: 'middle',
                platforms: {
                    ios: {
                        name: 'middle',
                        description: 'Middle #808080',
                        value:
                            'UIColor = UIColor(red: 0.502, green: 0.502, blue: 0.502, alpha: 1.0)',
                    },
                    android: {
                        name: 'middle',
                        value: '#808080',
                    },
                },
                format: 'color',
                type: 'string',
            },
            {
                id: 'white',
                platforms: {
                    ios: {
                        name: 'white',
                        description: 'White #FFFFFF',
                        value:
                            'UIColor = UIColor(red: 1.0, green: 1.0, blue: 1.0, alpha: 1.0)',
                    },
                    android: {
                        name: 'white',
                        value: '#FFFFFF',
                    },
                },
                format: 'color',
                type: 'string',
            },
        ],
    },
];
